src/main/java/com/mraof/minestuck/network/EditmodeDragPacket.java

```diff
@@ -0,0 +1,281 @@
+{
+	private static final Logger LOGGER = LogManager.getLogger();
+	
+	private static Tuple<Boolean, GristSet> editModePlaceCheck(EditData data, Player player, InteractionHand hand, BlockPos pos)
+	{
+		SburbConnection connection = data.getConnection();
+		ItemStack stack = player.getMainHandItem();
+		
+		if(stack.isEmpty() || hand.equals(InteractionHand.OFF_HAND) || !player.level.getBlockState(pos).getMaterial().isReplaceable())
+			return new Tuple<>(false, new GristSet());
+
+		DeployEntry entry = DeployList.getEntryForItem(stack, connection, player.level);
+		GristSet cost = entry != null ? entry.getCurrentCost(connection) : GristCost.findCostForItem(stack, null, false, player.level);
+		if(!GristHelper.canAfford(PlayerSavedData.getData(connection.getClientIdentifier(), player.level).getGristCache(), cost))
+			return new Tuple<>(false, cost);
+		
+		if(entry == null && !(stack.getItem() instanceof BlockItem))
+			return new Tuple<>(false, new GristSet());
+		
+		return new Tuple<>(true, new GristSet());
+	}
+	
+	private static Tuple<Boolean, GristSet> editModeDestroyCheck(EditData data, Player player, BlockPos pos)
+	{
+		SburbConnection connection = data.getConnection();
+		BlockState block = player.level.getBlockState(pos);
+		ItemStack stack = block.getCloneItemStack(null, player.level, pos, player);
+		DeployEntry entry = DeployList.getEntryForItem(stack, data.getConnection(), player.level, DeployList.EntryLists.ATHENEUM);
+		
+		if(block.isAir())
+			return new Tuple<>(false, new GristSet());
+		else if(!MinestuckConfig.SERVER.gristRefund.get() && entry == null)
+		{
+			GristSet cost = new GristSet(GristTypes.BUILD,1);
+			if(!GristHelper.canAfford(PlayerSavedData.getData(connection.getClientIdentifier(), player.level).getGristCache(), cost))
+				return new Tuple<>(false, cost);
+		}
+		return new Tuple<>(true, new GristSet());
+	}
+	
+	
+	public record Fill(boolean isDown, BlockPos positionStart, BlockPos positionEnd, Vec3 hitVector, Direction side) implements PlayToServerPacket
+	{
+		@Override
+		public void encode(FriendlyByteBuf buffer)
+		{
+			buffer.writeBoolean(isDown);
+			buffer.writeBlockPos(positionStart);
+			buffer.writeBlockPos(positionEnd);
+			buffer.writeDouble(hitVector.x);
+			buffer.writeDouble(hitVector.y);
+			buffer.writeDouble(hitVector.z);
+			buffer.writeEnum(side);
+		}
+		
+		public static Fill decode(FriendlyByteBuf buffer)
+		{
+			boolean isDragging = buffer.readBoolean();
+			BlockPos positionStart = buffer.readBlockPos();
+			BlockPos positionEnd = buffer.readBlockPos();
+			Vec3 hitVector = new Vec3(buffer.readDouble(), buffer.readDouble(), buffer.readDouble());
+			Direction side = buffer.readEnum(Direction.class);
+			
+			return new Fill(isDragging, positionStart, positionEnd, hitVector, side);
+		}
+		
+		@Override
+		public void execute(ServerPlayer player)
+		{
+			EditData data = ServerEditHandler.getData(player);
+			if(!player.getLevel().isClientSide() && data != null)
```

Comment:
`!player.getLevel().isClientSide()` should not be neccessary to check since this is in a function that should only be called server-side (the player being of class `ServerPlayer` should also tell us that we're server-side only already)

To reduce nesting, I recommend replacing the if-statement with:
```java
if(data == null)
	return;
```